hack/common.sh

```diff
@@ -862,4 +865,109 @@ function os::build::find-binary() {
   local path=$( (ls -t $(os::build::get-bin-output-path "${os_root}")/${bin}) 2>/dev/null || true | head -1 )
   echo "$path"
 }
-readonly -f os::build::find-binary
\ No newline at end of file
+readonly -f os::build::find-binary
+
+# os::build::environment::create creates a docker container with the default variables.
+# arguments are passed directly to the container, OS_BUILD_ENV_GOLANG, OS_BUILD_ENV_IMAGE,
+# and OS_RELEASE_DOCKER_ARGS can be used to customize the container. The docker socket
+# is mounted by default and the output of the command is the container id.
+function os::build::environment::create() {
+  set -o errexit
+  local golang_version="${OS_BUILD_ENV_GOLANG}"
+  local release_image="${OS_BUILD_ENV_IMAGE}"
+  local additional_context="${OS_BUILD_ENV_DOCKER_ARGS:-}"
+  if [[ -z "${additional_context}" && "${OS_BUILD_ENV_USE_DOCKER:-y}" == "y" ]]; then
+    additional_context="--privileged -v /var/run/docker.sock:/var/run/docker.sock"
+
+    if [[ "${OS_BUILD_ENV_LOCAL_DOCKER:-n}" == "y" ]]; then
+      # if OS_BUILD_ENV_LOCAL_DOCKER==y, add the local OS_ROOT as the bind mount to the working dir
+      # and set the running user to the current user
+      local workingdir
+      workingdir=$( os::build::environment::release::workingdir )
+      additional_context="${additional_context} -v ${OS_ROOT}:${workingdir} -u $(id -u)"
+    elif [[ -n "${OS_BUILD_ENV_REUSE_VOLUME:-}" ]]; then
+      # if OS_BUILD_ENV_REUSE_VOLUME is set, create a docker volume to store the working output so
+      # successive iterations can reuse shared code.
+      local workingdir
+      workingdir=$( os::build::environment::release::workingdir )
+      name="$( echo "${OS_BUILD_ENV_REUSE_VOLUME}" | tr '[:upper:]' '[:lower:]' )"
+      docker volume create --name "${name}" > /dev/null
+      additional_context="${additional_context} -v ${name}:${workingdir}"
+    fi
+  fi
+
+  # Create a new container to from the release environment
+  docker create ${additional_context} "${release_image}" "$@"
+}
+readonly -f os::build::environment::create
+
+# os::build::environment::release::workingdir calculates the working directory for the current
+# release image.
+function os::build::environment::release::workingdir() {
+  set -o errexit
```

Comment:
Ah, sorry, `set -o errtrace` should get the 'normal' behavior in our scripts `errexit` and `trap EXIT` get propagated. Make sure to install the stacktrace (`install_errexit`)